lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,291 @@
+          reportNotOnTop(order[0], shouldFirstNode, firstStatement)
+          return
+        }
+
+        // have only second and need to move it
+        if (shouldSecondNode && firstStatement !== shouldSecondNode) {
+          reportNotOnTop(order[1], shouldSecondNode, firstStatement)
+        }
+      }
+    }
+  )
+
+  /**
+   * @param {ASTNode} shouldFirstNode
+   * @param {ASTNode} shouldSecondNode
+   * @param {ASTNode} before
+   */
+  function reportBothNotOnTop(shouldFirstNode, shouldSecondNode, before) {
+    context.report({
+      node: shouldFirstNode,
+      loc: shouldFirstNode.loc,
+      messageId: 'macrosNotOnTop',
+      data: {
+        macro: order[0]
+      },
+      fix(fixer) {
+        return [
+          ...moveNodeBefore(fixer, shouldFirstNode, before),
+          ...moveNodeBefore(fixer, shouldSecondNode, before)
+        ]
+      }
+    })
+  }
+
+  /**
+   * @param {string} macro
+   * @param {ASTNode} node
+   * @param {ASTNode} before
+   */
+  function reportNotOnTop(macro, node, before) {
+    context.report({
+      node,
+      loc: node.loc,
+      messageId: 'macrosNotOnTop',
+      data: {
+        macro
+      },
+      fix(fixer) {
+        return moveNodeBefore(fixer, node, before)
+      }
+    })
+  }
+
+  /**
+   * Move all lines of "node" with its comments to before the "target"
+   * @param {RuleFixer} fixer
+   * @param {ASTNode} node
+   * @param {ASTNode} target
+   */
+  function moveNodeBefore(fixer, node, target) {
+    // get comments under tokens(if any)
+    const beforeNodeToken = sourceCode.getTokenBefore(node)
+    const nodeComment = sourceCode.getTokenAfter(beforeNodeToken, {
+      includeComments: true
+    })
+    const nextNodeComment = sourceCode.getTokenAfter(node, {
+      includeComments: true
+    })
+    // get positions of what we need to remove
+    const cutStart = getLineStartIndex(nodeComment, beforeNodeToken)
+    const cutEnd = getLineStartIndex(nextNodeComment, node)
+    // get space before target
+    const beforeTargetToken = sourceCode.getTokenBefore(target)
+    const targetComment = sourceCode.getTokenAfter(beforeTargetToken, {
+      includeComments: true
+    })
+    const textSpace = getTextBetweenTokens(beforeTargetToken, targetComment)
+    // make insert text: comments + node + space before target
+    const textNode = sourceCode.getText(
+      node,
+      node.range[0] - nodeComment.range[0]
+    )
+    const insertText = textNode + textSpace
+
+    return [
+      fixer.insertTextBefore(targetComment, insertText),
+      fixer.removeRange([cutStart, cutEnd])
+    ]
+  }
+
+  /**
+   * @param {ASTNode} tokenBefore
+   * @param {ASTNode} tokenAfter
+   */
+  function getTextBetweenTokens(tokenBefore, tokenAfter) {
+    const delta = tokenBefore.range[1] - tokenBefore.range[0]
+    const count = tokenAfter.range[0] - tokenBefore.range[1]
+
+    return sourceCode.getText(tokenBefore, -delta, count)
```

Comment:
It may be easier to get it directly from `text`.

```suggestion
    return sourceCode.text.slice(tokenBefore.range[1], tokenAfter.range[0])
```